librz/include/rz_util/rz_utf32.h

```diff
@@ -4,8 +4,16 @@
 /* For RzRune definition */
 #include "rz_utf8.h"
 
-RZ_API int rz_utf32_decode(const ut8 *ptr, int ptrlen, RzCodePoint *ch, bool bigendian);
+/**
+ * \brief Unicode byte order markings (BOM)
+ * according to https://www.unicode.org/faq/utf_bom.html#BOM
+ */
+#define RZ_UTF32_UNICODE_BOM_LE 0xFFFE0000
+#define RZ_UTF32_UNICODE_BOM_BE 0x0000FFFE
+
+RZ_API size_t rz_utf32_decode(RZ_NONNULL const ut8 *buf, size_t buf_len, RZ_NULLABLE RZ_OUT RzCodePoint *ch, bool big_endian);
 RZ_API int rz_utf32le_decode(const ut8 *ptr, int ptrlen, RzCodePoint *ch);
 RZ_API int rz_utf32be_decode(const ut8 *ptr, int ptrlen, RzCodePoint *ch);
+RZ_API bool rz_utf32_valid_cp(const ut8 *buf, size_t buf_len, bool big_endian);
```

Comment:
```suggestion
RZ_API bool rz_utf32_valid_codepoint(const ut8 *buf, size_t buf_len, bool big_endian);
```
Avoid using `cp` anywhere.